helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataWriter.java

```diff
@@ -77,8 +103,9 @@ public synchronized boolean addMetadataStoreRealm(String realm) {
       return createZkRealm(realm);
     }
 
-    // TODO: Forward the request to leader
-    return true;
+    String url_suffix = "namespaces/" + _namespace + "/metadata-store-realms/" + realm;
```

Comment:
1. Generally a good idea to refactor this to a private method so that it's easier to read. For instance, create `private String getUrlSuffix()`
2. Some of these string literals have already been defined in MetadataStoreDirectoryConstants.java. Could you use the constants there please?

Reply:
1. I wanted to include the url creation as a part of `forwardRequestToLeader`, but I do not see a good way to generalize url construction. The 4 endpoints require 2 different urls; refactoring it results in an if statement, which isn't generalization. On the other hand, leaving url construction logic in the methods also clearly indicates which urls each endpoints correspond to, which is good for maintenance. 
2. In `MetadataStoreRoutingConstants`, the constants are not quite the same as the strings in these urls. (`shardingKeys` vs `sharding-keys`, `metadataStoreRealms` vs `metadata-store-realms`)

Reply:
1. I do see a lot of duplication here. It seems it's possible to achieve the refactoring. Let me sync with you privately.

2. If you don't have them, you can create them. That's what the constant file is for. Please clearly note what they are used for.

Reply:
Resolved by offline discussion. A method that joins parameters will be used for url suffix building.